bbl_screen-patch/patches/printerui/qml/printer/SpeedAdjust.qml

```diff
@@ -0,0 +1,385 @@
+    property alias stepSize: dial.stepSize
+    property var stepSizes: [2, 10]
+    property var nearestLevel: function(val) { 
+        var closestIndex = 0; 
+        var minDiff = Math.abs(val - speedValues[0]);
+        for (var i = 1; i < speedValues.length; i++) {
+            var diff = Math.abs(val - speedValues[i]);
+            if (diff < minDiff) {
+                minDiff = diff; 
+                closestIndex = i;
+            }
+        }
+        return 3 - closestIndex; 
+    }
+
+    onTargetChanged:{
+        if (target != null){
+            dial.value = currentSpeed;
+            dialCanvas.requestPaint();
+            curValArc.requestPaint();
+        }
+    }
+    Component.onCompleted: {
+        if (target != null){
+            stepSize = X1Plus.Settings.get("printerui.speedadjust.stepSize", 2);
+            dial.value = currentSpeed;
+            dialCanvas.requestPaint();
+            curValArc.requestPaint();
+        }
+    }
+    
+    /* Button for switching to stepbar */
+    Rectangle {
+        id: toStepbar
+        width: 90
+        height: 90
+        color: Colors.gray_600
+        x: 160
+        y: 20
+        z:1
+        visible: customSpeed
+        
+        ZImage {
+            id: defaultButton
+            width: 60
+            height: 60
+            anchors.centerIn: parent
+            fillMode: Image.PreserveAspectFit 
+            originSource:  "../../icon/right.svg"
+            transform: Scale {
+                origin.x: defaultButton.width / 2
+                origin.y: defaultButton.height / 2
+                xScale: customSpeed ? -1 : 1
+            }
+        }
+
+        MouseArea {
+            anchors.fill: parent
+            onClicked: {
+                customSpeed = false;
+                X1Plus.Settings.put("printerui.speedadjust", customSpeed);
+                /*  set the stepbar's value to the nearest OEM speed profile */
+                let equivLevel = nearestLevel(currentSpeed);
+                stepBar.step = equivLevel;
+                PrintManager.currentTask.printSpeedMode = equivLevel;
+            }
+            onEntered: parent.opacity = 0.8
+            onExited: parent.opacity = 1.0
+        }
+        
+    }
+
+    Rectangle {
+        id: dialRect
+        width: 540
+        height: 591
+        anchors.centerIn: parent
+        radius: 15
+        color: Colors.gray_600
+        MouseArea {
+            anchors.fill: parent
+            onPressed: mouse.accepted = true
+        }
+        Text {
+            id: title
+            text:qsTr("Print Speed")
+            font.pixelSize: 36
+            color: "white"
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.top: parent.top
+            anchors.topMargin: 40
+        }
+        Dial {
+            id: dial
+            width: 300
+            height: width
+            from: 30
+            stepSize:2
+            to:180
```

Comment:
This is 50 to 166 now, since we don't extrapolate outside of the Bambu domain, right?